### DISCUSSION TITLE
INDRA script generation

### DISCUSSION THREAD

**Discussion by xandmac** [churchmanlab/genewalk#57] [Q&A] [answered]
Hello, I was looking to explore creating a custom INDRA input. I was wondering if you could provide the script used to create the INDRA network from the paper. 

Thanks!

**Comment by bgyori** [comment, marked as answer]
Hi @xandmac, sorry for the slow response. First of all, here are three Python pickle files that contain INDRA statements for the experiments in the paper:
https://bigmech.s3.amazonaws.com/genewalk/IsoG_HGNCidForINDRA.pkl
https://bigmech.s3.amazonaws.com/genewalk/QKI_MGIforINDRA_stmts.pkl
https://bigmech.s3.amazonaws.com/genewalk/JQ1_HGNCidForINDRA.pkl

In terms of building your own INDRA network, that can take additional work but it gives you flexibility in what kind of relations you want to include from multiple sources. You can see an overview of the knowledge sources that INDRA makes available here: https://github.com/sorgerlab/indra#knowledge-sources, for instance, you might process BioGRID and Signor to get Statements, save it as a pickle file, and pass that to GeneWalk as the `--network_file`. You can also combine this with text mining through one or more of the systems INDRA is integrated with, for instance, Reach. While this takes additional effort to set up, it allows you to derive Statements from subsets of literature that you determine as relevant, and you may relations that may not be available from pathway databases.

**Reply by ATPs** [reply]
Hi, @bgyori . It seems that the pickle files you mentioned are for particular genes, so we cannot use it directly in our own project. I went through INDRA, and found that it runs for each specific gene? Is it possible to create a network for all genes like pathwaycommons, with just some resources? It seems hard to do...

**Reply by bgyori** [reply]
Hi @ATPs, let me try to clarify. INDRA is integrated with multiple text mining systems and pathway database sources. Most pathway database sources (e.g. Signor, BioGRID, etc.) are integrated such that there is a function to process the entire resource into INDRA Statements - in this sense one doesn't have to process the content gene by gene. As for text mining, similarly, if you identify a set of papers and process them using a text mining system, all statements extracted from text (not just ones for a specific gene) are returned. You can process or filter the statements obtained and create a custom network for your purposes.

**Comment by xandmac** [comment]
Oh thanks for providing the scripts. Very helpful!
